Content.Shared/Projectiles/ProjectileComponent.cs

```diff
@@ -73,4 +79,12 @@ public sealed partial class ProjectileComponent : Component
     /// </summary>
     [DataField]
     public bool DamagedEntity;
+
+    /// <summary>
+    /// Sets the maximum range for a projectile fired with ShootAtFixedPointComponent.
+    /// This can be set on both the Projectile and ShootAtFixedPoint Components.
+    /// The default value is null for no cap. The minimum value between the two is used.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public float? MaxFixedRange;
```

Comment:
I added a new component that should be able to replace this with the Boiler PR https://github.com/RMC-14/RMC-14/pull/3258/files, ProjectileMaxRangeComponent